### PR TITLE
Add configuration options to fetch admin passwords from files

### PR DESCRIPTION
**Pull request by kriswuollett**
**Description of the change**

Adds `LDAP_ADMIN_PASSWORD_FILE` and `LDAP_CONFIG_ADMIN_PASSWORD_FILE` environment variable options to specify files from which the encrypted passwords should be generated instead of `LDAP_ADMIN_PASSWORD_FILE` 
 and `LDAP_CONFIG_ADMIN_PASSWORD_FILE` respectively.

**Benefits**

- Allows the use of mounted secrets files in a container environment for sourcing unencrypted passwords.
- The unencrypted password is not available within environment variables of the process.

**Possible drawbacks**

Makes password initialization more opaque since in general there is not a lot of `info`-level logging during the initialization process.

**Applicable issues**

- Partially completes the features requested in #62.

**Additional information**

The implementation approach is done with minimal changes to the existing codebase. It could be improved later by refactoring out the encrypted password logic and adding debug logging to inform the operator what options are being used.

### CODE REVIEW

File: 2.4/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh [hunk 1]
```diff
@@ -50,12 +50,14 @@ export LDAP_ROOT="${LDAP_ROOT:-dc=example,dc=org}"
 export LDAP_ADMIN_USERNAME="${LDAP_ADMIN_USERNAME:-admin}"
 export LDAP_ADMIN_DN="${LDAP_ADMIN_USERNAME/#/cn=},${LDAP_ROOT}"
 export LDAP_ADMIN_PASSWORD="${LDAP_ADMIN_PASSWORD:-adminpassword}"
-export LDAP_ENCRYPTED_ADMIN_PASSWORD="$(echo -n $LDAP_ADMIN_PASSWORD | slappasswd -n -T /dev/stdin)"
+export LDAP_ADMIN_PASSWORD_FILE="${LDAP_ENCRYPTED_ADMIN_PASSWORD:-}"
+export LDAP_ENCRYPTED_ADMIN_PASSWORD="$(([[ -f "$LDAP_ADMIN_PASSWORD_FILE" ]] && cat "$LDAP_ADMIN_PASSWORD_FILE" || echo -n $LDAP_ADMIN_PASSWORD) | slappasswd -n -T /dev/stdin)"
```
Comment (juan131) [inline 2.4/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:54]: Have you considered using [the approach we use in MariaDB](https://github.com/bitnami/bitnami-docker-mariadb/blob/master/10.6/debian-10/rootfs/opt/bitnami/scripts/mariadb-env.sh)? Sth like this

```bash
openldap_env_vars=(
     LDAP_ADMIN_PASSWORD
     LDAP_CONFIG_ADMIN_PASSWORD
)
for env_var in "${openldap_env_vars[@]}"; do
    file_env_var="${env_var}_FILE"
    if [[ -n "${!file_env_var:-}" ]]; then
        if [[ -r "${!file_env_var:-}" ]]; then
            export "${env_var}=$(< "${!file_env_var}")"
            unset "${file_env_var}"
        else
            warn "Skipping export of '${env_var}'. '${!file_env_var:-}' is not readable."
        fi
    fi
done
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please check my comments

File: 2.4/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh (hunk 1 above)
Comment (kriswuollett) [reply 2.4/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:54]: Did not look at your other container repos yet. Thanks for pointing it out. Would be great to find something that would both bring the container closer to something that I'd use along with code I'd be happy to put in the overhead of testing to get it committed, or at least shared code we'd both benefit from.

If there is an existing solution in place in another bitnami repo that you distribute, then just copying it here may not be worth the overhead for me as I've already set up a basic container with Alpine to get #65 as well as #63. On the other hand a stepwise improvement would be to extract that logic for an individual environment variable into something like [`libvalidations.sh:is_int`](https://github.com/bitnami/bitnami-docker-openldap/blob/b1f54da9cd0dc4e190afde6e1d7acdfd24f7ab2a/2.5/debian-10/prebuildfs/opt/bitnami/scripts/libvalidations.sh#L19) would be interesting. That's possible, right? Where is the upstream repo for `prebuildfs/opt/bitnami/scripts`?

Comment (juan131) [reply 2.4/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:54]: Hi @kriswuollett 

The repo containing the "source of truth" for every Bitnami container isn't public and it's quite complex. In any case, consider these scripts are maintained directly in this repo. We usually do the review directly through these `bitnami-docker-XXX` repositories and, after changes are approved/merged, then we translate them to this private repo.

**Pull request closed without merging**